src/jaxsim/api/model.py

```diff
@@ -2056,7 +2111,7 @@ def step(
             jaxsim.exceptions.raise_runtime_error_if(
                 condition=jnp.logical_and(
                     isinstance(
-                        integrator,
+                        model.integrator,
```

Comment:
My bad, we should leave `integrator` here as it is the one used for stepping the simulation 
```suggestion
                        integrator,
```

Reply:
Done in https://github.com/ami-iit/jaxsim/pull/252/commits/68410809a1454a7225d52d8f4e09940ff69c7e13